stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -670,13 +670,31 @@ function opnorm1(A::AbstractMatrix{T}) where T
     return convert(Tnorm, nrm)
 end
 
+# Uses power iteration to compute the maximal singular value.
+# falls back to svdvals if it runs into numerics issues.
 function opnorm2(A::AbstractMatrix{T}) where T
     require_one_based_indexing(A)
     m,n = size(A)
     Tnorm = typeof(float(real(zero(T))))
     if m == 0 || n == 0 return zero(Tnorm) end
     if m == 1 || n == 1 return norm2(A) end
-    return svdvals(A)[1]
+    # to minimize the chance of x being orthogonal to the largest eigenvector
+    x = randn(Tnorm, n)
+    tmp = zeros(Tnorm, m)
+    At = A'
+    v = one(Tnorm)
+    # this will converge quickly as long as the top two eigenvalues are distinct
```

Comment:
Couldn't we use a Lanczos iteration to be more robust and converge quickly even if a few eigenvalues are similar in magnitude?

Reply:
This paper seems to be a popular algorithm: https://doi.org/10.1137/04060593X

Reply:
good point. There's a python implementation of that paper with a good license here: https://github.com/bwlewis/irlbpy/blob/master/irlb/irlb.py. I'll see how it does.

Reply:
good news and bad news. Good news is I have it working. Bad news is it takes about 84 lines of code. Performance testing to come.